pulp_2_tests/tests/rpm/api_v2/test_modularity.py

```diff
@@ -200,3 +260,68 @@ def test_all(self):
             with self.subTest(package=key):
                 module = [line for line in lines if key in line]
                 self.assertEqual(len(module), value, module)
+
+
+class UploadModuleTestCase(unittest.TestCase):
+    """Upload a module.yaml file and test upload import in Pulp repo."""
+
+    def test_all(self):
+        """Verify whether uploaded module.yaml is reflected in the pulp repo."""
+        cfg = config.get_config()
+        if cfg.pulp_version < Version('2.17'):
+            raise unittest.SkipTest('This test requires at least Pulp 2.17 or newer.')
+        if not os_support_modularity(cfg):
+            raise unittest.SkipTest('This test requires an OS that supports modularity.')
+        client = api.Client(cfg, api.json_handler)
+        # Create a normal Repo without any data.
+        body = gen_repo(
+            importer_config={'feed': RPM_SIGNED_FEED_URL},
+            distributors=[gen_distributor()]
+        )
+        repo = client.post(REPOSITORY_PATH, body)
+        repo = client.get(repo['_href'], params={'details': True})
+        self.addCleanup(client.delete, repo['_href'])
+        sync_repo(cfg, repo)
+
+        # download modlues.yaml and upload it to pulp_repo
+        unit = self._get_module_xml_file(cfg, RPM_WITH_MODULES_FEED_URL)
+        upload_import_unit(cfg, unit, {
+            'unit_key': {},
+            'unit_type_id': 'modulemd',
+        }, repo)
+        repo = client.get(repo['_href'], params={'details': True})
+        # Assert that `modulemd` and `modulemd_defaults` are present on the
+        # repository.
+        self.assertIsNotNone(repo['content_unit_counts']['modulemd'])
+        self.assertIsNotNone(repo['content_unit_counts']['modulemd_defaults'])
+
+    @staticmethod
+    def _get_module_xml_file(cfg, path):
```

Comment:
Look into [modules.yaml.gz](https://repos.fedorapeople.org/pulp/pulp/fixtures/rpm-with-modules/repodata/) there is no `modules.xml` neither `modules.xml.gz`. Lets re-evaluate the use of this function.